src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -127,6 +133,78 @@ def __getattribute__(self, name):
         return val
 
 
+def in_cloud_console():
+    return os.environ.get('ACC_CLOUD', None)
+
+
+def _ssl_context():
+    if sys.version_info < (3, 4) or (in_cloud_console() and platform.system() == 'Windows'):
+        try:
+            return ssl.SSLContext(ssl.PROTOCOL_TLS)  # added in python 2.7.13 and 3.6
+        except AttributeError:
+            return ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+
+    return ssl.create_default_context()
+
+
+def urlretrieve(url):
+    req = urlopen(url, context=_ssl_context())
+    return req.read()
```

Comment:
This is duplication code copied and pasted from other place, please refactor

Reply:
Done, moved to util.py